### PR TITLE
Fix crash while trying to autocomplete non-global Autoload

### LINKED ISSUE #63031: Code Autocomplete causes crash when referencing Autoload script with disabled Global Variable toggle.
### Godot version

4.0 alpha 12 x64

### System information

Windows 10, x86_64, Vulcan Desktop v1.2.0, Ryzen 5 1600, RTX3070 (516.59 Studio Driver)

### Issue description

If you set an autoload with "Global Variable" enabled, you can access it anywhere with the name of the autoload directly.

If you _disable_ the "Global Variable" toggle and attempt or accidentally reference the Autoload via the same naming and type a single period EG:`ExampleAutoload.` which would normally prompt code autocompletion, **instead crashes the engine** with a single line of error output (if running Godot with a command line debug console)

- ERROR: FATAL: Condition "!exists" is true.
 at: operator[] (./core/templates/hash_map.h:504)
not exists

For now I'll just try to avoid having Autoload scripts that are _not_ also declared as Global Variables, but this crash seems like explicitly undesired behaviour.
Error also seems to be present in Alpha 11

### Steps to reproduce

1. Create a new project.
2. Create a new script, inherit from node. Name it but don't change it, just save it.
3. Set that script as an Autoload script and _disable_ "Global Variable" toggle.
4. Create any new script, and type the case-sensitive name of the Autoload script immediately followed by a period.
5. Crash.

### Minimal reproduction project

[AutoloadRefCrash.zip](https://github.com/godotengine/godot/files/9120310/AutoloadRefCrash.zip)
Project has two Autoloads (ExampleAutoload and AutoloadTwo, referencing the former will not crash and will show code completion output, referencing the latter will instantly crash the engine)

### PR DESCRIPTION
**Pull request by cdemirer**
Fixes #63031

### CODE REVIEW

File: modules/gdscript/gdscript_editor.cpp [hunk 1]
```diff
@@ -2005,7 +2005,7 @@ static bool _guess_identifier_type(GDScriptParser::CompletionContext &p_context,
 	}
 
 	// Check autoloads.
-	if (ProjectSettings::get_singleton()->has_autoload(p_identifier)) {
+	if (ProjectSettings::get_singleton()->has_autoload(p_identifier) && GDScriptLanguage::get_singleton()->get_named_globals_map().has(p_identifier)) {
```
Comment (nathanfranke) [inline modules/gdscript/gdscript_editor.cpp:2008]: I suggest a comment explaining why `get_named_globals_map()` may not contain `p_identifier`.

```suggestion
	// Check autoloads. Identifier may not be in the globals map if "Global Variable" is disabled.
	if (ProjectSettings::get_singleton()->has_autoload(p_identifier) && GDScriptLanguage::get_singleton()->get_named_globals_map().has(p_identifier)) {
```

Note: Do not apply my changes through GitHub, as I could not replace the previous comment.

File: modules/gdscript/gdscript_editor.cpp [hunk 2]
```diff
@@ -2004,8 +2004,8 @@ static bool _guess_identifier_type(GDScriptParser::CompletionContext &p_context,
 		return false;
 	}
 
-	// Check autoloads.
-	if (ProjectSettings::get_singleton()->has_autoload(p_identifier)) {
+	// Check autoloads. Identifier may not be in the named globals map if "Global Variable" option is disabled.
+	if (ProjectSettings::get_singleton()->has_autoload(p_identifier) && GDScriptLanguage::get_singleton()->get_named_globals_map().has(p_identifier)) {
```
Comment (vnen) [inline modules/gdscript/gdscript_editor.cpp:2008]: If you're checking the `get_named_globals_map()` you don't need to check the autoloads in ProjectSettings, it's redundant (named globals are only used for autoloaded singletons anyway).

**Review by vnen — DISMISSED** [review]

File: modules/gdscript/gdscript_editor.cpp (hunk 2 above)
Comment (nathanfranke) [reply modules/gdscript/gdscript_editor.cpp:2008]: Will we ever implement globals that aren't Autoloads?

Comment (akien-mga) [reply modules/gdscript/gdscript_editor.cpp:2008]: It probably doesn't matter in that case, the "autoload" part of the comment could be replaced by "global variables".

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

File: modules/gdscript/gdscript_editor.cpp (hunk 2 above)
Comment (nathanfranke) [reply modules/gdscript/gdscript_editor.cpp]: Looks great now :slightly_smiling_face:

**Pull request merged**